services/terraformer/templates/aws.tpl

```diff
@@ -0,0 +1,160 @@
+
+provider "aws" {
+  region     = "{{(index .NodePools 0).Region}}"
+  access_key = "{{(index .NodePools 0).Provider.AccessKey}}"
+  secret_key = file("{{(index .NodePools 0).Provider.SpecName}}")
+  alias = "k8s-nodepool"
+}
+
+resource "aws_vpc" "claudie-vpc" {
+  provider = aws.k8s-nodepool
+  cidr_block = "10.0.0.0/16"
+  tags = {
+    Name = "{{ $clusterName }}-{{ $clusterHash }}-vpc"
+  }
+}
+
+resource "aws_subnet" "claudie-subnet" {
+  provider = aws.k8s-nodepool
+  vpc_id            = aws_vpc.claudie-vpc.id
+  cidr_block        = "10.0.0.0/24"
+  map_public_ip_on_launch = true
+  availability_zone = "{{(index .NodePools 0).Zone}}"
+  tags = {
+    Name = "{{ $clusterName }}-{{ $clusterHash }}-subnet"
+  }
+}
+
+resource "aws_internet_gateway" "claudie-gateway" {
+  provider = aws.k8s-nodepool
+  vpc_id = aws_vpc.claudie-vpc.id
+  tags = {
+    Name = "{{ $clusterName }}-{{ $clusterHash }}-gateway"
+  }
+}
+
+resource "aws_route_table" "claudie-route-table" {
+  provider = aws.k8s-nodepool
+  vpc_id = aws_vpc.claudie-vpc.id
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.claudie-gateway.id
+  }
+  tags = {
+    Name = "{{ $clusterName }}-{{ $clusterHash }}-rt"
+  }
+}
+
+resource "aws_route_table_association" "claudie-rta" {
+  provider = aws.k8s-nodepool
+  subnet_id = aws_subnet.claudie-subnet.id
+  route_table_id = aws_route_table.claudie-route-table.id
+}
+
+resource "aws_security_group" "claudie-sg" {
+  provider = aws.k8s-nodepool
+  vpc_id      = aws_vpc.claudie-vpc.id
+  revoke_rules_on_delete = true
+  tags = {
+    Name = "{{ $clusterName }}-{{ $clusterHash }}-sg"
+  }
+}
+
+resource "aws_security_group_rule" "allow-egress" {
+  provider = aws.k8s-nodepool
+  type              = "egress"
+  from_port         = 0
+  to_port           = 65535
+  protocol          = "-1"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.claudie-sg.id
+}
+
+
+resource "aws_security_group_rule" "allow-ssh" {
+  provider = aws.k8s-nodepool
+  type              = "ingress"
+  from_port         = 22
+  to_port           = 22
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.claudie-sg.id
+}
+
+resource "aws_security_group_rule" "allow-kube-api" {
+  provider = aws.k8s-nodepool
+  type              = "ingress"
+  from_port         = 6443
+  to_port           = 6443
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.claudie-sg.id
+}
+
+
+resource "aws_security_group_rule" "allow-wireguard" {
+  provider = aws.k8s-nodepool
+  type              = "ingress"
+  from_port         = 51820
+  to_port           = 51820
+  protocol          = "udp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.claudie-sg.id
+}
+
+resource "aws_security_group_rule" "allow-icmp" {
+  provider = aws.k8s-nodepool
+  type              = "ingress"
+  from_port         = -1
```

Comment:
This could be hardened to only allow `ping`, but I don't know if it's important or worth doing.
Might as well, though.
https://blog.jwr.io/terraform/icmp/ping/security/groups/2018/02/02/terraform-icmp-rules.html